### PR TITLE
Fix hwmap error

### PR DESCRIPTION
**Pull request by wenbinc-Bin**
There are three teams bothered by this issue. Temporarily submit to carthwheel until they are merged into upstream.

### CODE REVIEW

**Comment by xhaihao** [comment]
@wenbinc-Bin Please change the prefix number in the patch names.  BTW 0097 is also required for OCL issue, right ? I thought only 0098 is required.

**Comment by wenbinc-Bin** [comment]
Only 0097 is required for OCL issue. They are on the same upstream patchset, so I submit them together here.

File: patches/0092-libavutil-hwcontext_opencl-fix-a-bug-for-mapping-qsv.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+--- a/libavutil/hwcontext_qsv.h
++++ b/libavutil/hwcontext_qsv.h
+@@ -29,6 +29,11 @@
+  * contain AVBufferRefs whose data pointer points to an mfxFrameSurface1 struct.
+  */
+ 
++#if CONFIG_VAAPI
++#define MFXSURFACEP_TO_VASURFACEP(surf) \
++    (VASurfaceID*)(((mfxHDLPair*)surf->Data.MemId)->first)
++#endif
```
Comment (xhaihao) [inline patches/0092-libavutil-hwcontext_opencl-fix-a-bug-for-mapping-qsv.patch:56]: Is it possible to remove the dependency on VAAPI in a public header file ?

**Comment by wenbinc-Bin** [comment]
Let's do this on mail-list. close this pr.

**Pull request closed without merging**